### PR TITLE
Refactor atom.dart to make code cleaner

### PR DESCRIPTION
**Pull request by fzyzcjy**
Describe the changes proposed in this Pull Request.

If the PR fixes a specific issue, reference the issue with **`Fixes #`**.

---
## Pull Request Checklist

- [x] If the changes are being made to code, ensure the **version in `pubspec.yaml`** is updated. 
- [x] Increment the **`major`/`minor`/`patch`/`patch-count`**, depending on the complexity of change
- [x] Add the necessary **unit tests** to ensure the coverage does not drop
- [x] Update the **Changelog** to include all changes made in this PR
- [x] Run the **`set:versions` command** using `npm` or `yarn`. You can find this command in the `package.json` file in the root directory
- [x] Include the **necessary reviewers** for the PR
- [x] Update the docs if there are any API changes or additions to functionality

### CODE REVIEW

**Comment by pavanpodila** [comment]
Why is the code coverage dropping here? Can you please check

**Comment by fzyzcjy** [comment]
No idea... It looks equivalent indeed

image

**Review by amondnet — APPROVED** [review]
LGTM

**Review by pavanpodila — CHANGES_REQUESTED** [review]
LGTM but need version change on pubspec, NPM set:versions and changelog

File: mobx/lib/src/core/atom.dart [hunk 1]
```diff
@@ -96,11 +96,7 @@ class Atom {
       _addListener(_ListenerKind.onBecomeUnobserved, fn);
 
   void Function() _addListener(_ListenerKind kind, void Function() fn) {
-    if (_observationListeners[kind] == null) {
-      _observationListeners[kind] = {}..add(fn);
-    } else {
-      _observationListeners[kind]!.add(fn);
-    }
+    (_observationListeners[kind] ??= {}).add(fn);
```
Comment (pavanpodila) [inline mobx/lib/src/core/atom.dart:99]: will need to add a version change to publish this

**Review by pavanpodila — CHANGES_REQUESTED** [review]

**Comment by ghenry** [comment]
Hi all,

This looks good to merge and close?

Thanks.

File: mobx/lib/src/core/atom.dart (hunk 1 above)
Comment (amondnet) [reply mobx/lib/src/core/atom.dart:99]: @pavanpodila done

**Comment by amondnet** [comment]
@all-contributors please add @amondnet for review

**Pull request merged**